docs/quick_start/quick_start_with_jdbc.md

```diff
@@ -15,82 +15,76 @@
- 
-public class JDBCTest {
- 
-    private static String driverName = "org.apache.hive.jdbc.HiveDriver";
-    private static String kyuubiJdbcUrl = "jdbc:hive2://localhost:10009/default;";
- 
-    public static void main(String[] args) throws ClassNotFoundException, SQLException {
-        String principal = args[0]; // kerberos principal
-        String keytab = args[1]; // keytab file location
-        Configuration configuration = new Configuration();
-        configuration.set(HADOOP_SECURITY_AUTHENTICATION, "kerberos");
-        UserGroupInformation.setConfiguration(configuration);
-        UserGroupInformation ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal, keytab);
- 
-        Class.forName(driverName);
-        Connection conn = ugi.doAs(new PrivilegedExceptionAction<Connection>(){
-            public Connection run() throws SQLException {
-                return DriverManager.getConnection(kyuubiJdbcUrl);
-            }
-        });
-        Statement st = conn.createStatement();
-        ResultSet res = st.executeQuery("show databases");
-        while (res.next()) {
-            System.out.println(res.getString(1));
+public class KyuubiJDBC {
+
+    private static String driverName = "org.apache.kyuubi.jdbc.KyuubiHiveDriver";
+    private static String kyuubiJdbcUrl = "jdbc:kyuubi://localhost:10009/default;";
+
+    public static void main(String[] args) throws SQLException {
+        Connection conn = DriverManager.getConnection(kyuubiJdbcUrl);
+        Statement stmt = conn.createStatement();
+        ResultSet rs = st.executeQuery("show databases");
+        while (rs.next()) {
+            System.out.println(rs.getString(1));
         }
-        res.close();
-        st.close();
+        rs.close();
+        stmt.close();
         conn.close();
     }
 }
 ```
+## Connect to Kerberized Kyuubi Server
+
+The following Java code is using a keytab file to login and connect to Kyuubi Server by JDBC.
 
+```java
+package org.apache.kyuubi.examples;
+import java.sql.*;
+public class KyuubiJDBCDemo {
+    private static String driverName = "org.apache.kyuubi.jdbc.KyuubiHiveDriver";
+    private static String kyuubiJdbcUrlTemplate = "jdbc:kyuubi://localhost:10009/default;" +
+            "clientPrincipal=%s;clientKeytab=%s;serverPrincipal=%s";
+    public static void main(String[] args) throws SQLException {
+        String clientPrincipal = args[0]; // Kerberos principal
+        String clientKeytab = args[1];    // Keytab file location
+        String serverPrincipal = arg[2];  // Kerberos principal used by Kyuubi Server
+        String kyuubiJdbcUrl = String.format(kyuubiJdbcUrl, clientPrincipal, clientKeytab, serverPrincipal);
+        Connection conn = DriverManager.getConnection(kyuubiJdbcUrl);
+        Statement stmt = conn.createStatement();
+        ResultSet rs = st.executeQuery("show databases");
+        while (rs.next()) {
+            System.out.println(rs.getString(1));
+        }
+        rs.close();
```

Comment:
nit: better in `finally` and wrap each close with separated `try-finally` blocks

Reply:
updated